snippets/perl/migration.pm

```diff
@@ -0,0 +1,201 @@
+    This subroutine is used to call /security/certificate-authorities/activate api to
+    activate the new CA in migration process.
+ PARAMETERS
+    @param $_[0] - string
+        The name of the master server.
+        Ex: myhostname.myDomain.com
+    @param $_[1] - string
+        The authentication token fetched after invoking /login api using 
+        user credentials or apikey.
+    @param $_[2] - string - optional
+        A textual description to activate the new CA.
+    @param $_[3] - int [0/1] - optional
+        Forcefully activate the new CA.
+		  
+ RETURNS
+    None
+=cut
+
+sub activate_migration {
+    my @argument_list = @_;
+    my $force = 0;
+    my $reason;
+    my $base_url = $argument_list[0];
+    my $token = $argument_list[1];
+    if (@argument_list == 4) {
+        $reason = $argument_list[2];
+        $force = $argument_list[3];
+    } elsif (@argument_list == 3) {
+        $force = $argument_list[2];
+    }
+
+    my $url = "$base_url/security/certificate-authorities/activate";
+    my $req = HTTP::Request->new(POST => $url);
+    $req->header('Content-Type' => $MIGRATION_NB_CONTENT_TYPE_V4);
+    $req->header('Authorization' => $token);
+    $req->header('X-NetBackup-Audit-Reason' => $reason) if (defined $reason);
+    my $post_data;
+    if ($force) {
+        $post_data = qq({ "data": { "type": "nbcaMigrationActivateRequest", "attributes": {
+                         "force": "true" } } });
+    }
+    else {
+        $post_data = qq({ "data": { "type": "nbcaMigrationActivateRequest", "attributes": { }}});
+    }
+
+    $req->content($post_data);
+
+    print "\n**************************************************************";
+    print "\n Making POST Request to activate the new CA \n";
+
+    my $resp = $ua->request($req);
+    if ($resp->is_success) {
+         print "Successfully activated the new CA with status code: ", $resp->code, "\n";
+    } else {
+         printErrorResponse($resp);
+    }
+}
+
+=head1 complete_migration
+ SYNOPSIS
+    This subroutine is used to call /security/certificate-authorities/complete-migration api to
+    complete the CA migration process.
+ PARAMETERS
+    @param $_[0] - string
+        The name of the master server.
+        Ex: myhostname.myDomain.com
+    @param $_[1] - string
+        The authentication token fetched after invoking /login api using 
+        user credentials or apikey.
+    @param $_[2] - string - optional
+        A textual description to complete the CA migration process.
+    @param $_[3] - int [0/1] - optional
+        Forcefully complete the CA migration process.
+		  
+ RETURNS
+    None
+=cut
+
+sub complete_migration {
+    my @argument_list = @_;
+    my $force = 0;
+    my $reason;
+    my $base_url = $argument_list[0];
+    my $token = $argument_list[1];
+
+    if (@argument_list == 4) {
+        $reason = $argument_list[2];
+        $force = $argument_list[3];
+    } elsif (@argument_list == 3) {
+        $force = $argument_list[2];
+    }
```

Comment:
Parameters 2 and 3 are both listed as optional. If I provide parameter 2 and not parameter 3, then the length of the array will be 3, but this will assign the value to `$force` instead of to `$reason`. I think you can simplify the whole thing with a single line:

```perl
my ($base_url, $token, $reason, $force) = @_;